code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -1,67 +1,45 @@
 import React from 'react';
 import { STAGE_PROGRESS_TYPE } from './types';
-import { saveAnswersAndNavigate } from '../../levels/saveAnswers';
+import ProgressDot from './progress_dot.jsx';
+import color from '../../color';
+
+const STYLES = {
+  courseOverviewContainer: {
+    display: 'table-cell',
+    verticalAlign: 'middle',
+    paddingRight: 10
+  },
+  headerContainer: {
+    padding: '5px 8px',
+    backgroundColor: color.lightest_gray,
+    border: `1px solid ${color.lighter_gray}`,
+    borderRadius: 5
+  }
+};
 
 /**
  * Stage progress component used in level header and course overview.
  */
-var StageProgress = React.createClass({
+let StageProgress = React.createClass({
   propTypes: {
     levels: STAGE_PROGRESS_TYPE,
-    currentLevelIndex: React.PropTypes.number,
+    currentLevelId: React.PropTypes.string,
     largeDots: React.PropTypes.bool,
     saveAnswersFirst: React.PropTypes.bool.isRequired
   },
 
-  dotClicked(url) {
-    if (saveAnswersAndNavigate) {
-      saveAnswersAndNavigate(url);
-    }
-  },
-
   render() {
-    var progressDots = this.props.levels.map((level, index) => {
-
-      var innerClass = 'level_link ' + (level.status || 'not_tried');
-      if (level.kind === 'unplugged') {
-        innerClass += ' unplugged_level';
-      }
-
-      var outerClass = (level.kind === 'assessment') ? 'puzzle_outer_assessment' : 'puzzle_outer_level';
-      outerClass = (index === this.props.currentLevelIndex) ? 'puzzle_outer_current' : outerClass;
-
-      var isUnplugged = isNaN(level.title);
-      var dotStyle = {};
-      if (this.props.largeDots) {
-        if (isUnplugged) {
-          dotStyle = {fontSize: '13px', padding: '4px 10px', margin: '1px'};
-        } else {
-          dotStyle = {padding: '5px 4px 3px 4px', margin: '1px'};
-        }
-      }
-
-      var onClick = null;
-      if (this.props.saveAnswersFirst) {
-        dotStyle.cursor = 'pointer';
-        onClick = (e) => {this.dotClicked(level.url); e.preventDefault();};
-      }
-
-      return ([
-        <div className={outerClass}>
-          <a
-            href={level.url}
-            onClick={onClick}
-            className={innerClass + ' level-' + level.id}
-            style={dotStyle}>
-              {level.title}
-          </a>
-        </div>,
-        ' '
-      ]);
-    });
+    let progressDots = this.props.levels.map(level => ([
+      <ProgressDot
+        level={level}
+        currentLevelId={this.props.currentLevelId}
+        largeDots={this.props.largeDots}
+        saveAnswersFirst={this.props.saveAnswersFirst}
+      />, ' '
+    ]));
```

Comment:
Gif showing the difference with margin vs. [spacing from white space](http://stackoverflow.com/a/8394652):
![dot_spacing](https://cloud.githubusercontent.com/assets/413693/15842341/ffd6a040-2c0c-11e6-8be3-a84ebcfcfa57.gif)

White spacing:
![before](https://cloud.githubusercontent.com/assets/413693/15842344/0e68b08a-2c0d-11e6-8846-7d9c92007e3d.png)

Margin:
![after](https://cloud.githubusercontent.com/assets/413693/15842347/13c91a9c-2c0d-11e6-9f90-8ec2f3862b49.png)

I think the diff is acceptable so I'll go ahead and remove `' '` and add the margin.  It may be a small enough change to not trigger an Eyes test failure.